### PR TITLE
 VGG19 Implementation with example 

### PR DESCRIPTION
**Pull request by OwenLiuzZ**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by DEKHTIARJonathan** [comment]
@OwenLiuzZ Thank you so much for your PR.

Please update the changelog with your contribution and add your name to the contributor list.
**Please follow the formating**

I let @zsdonghao reviewing your work ;)

**Comment by zsdonghao** [comment]
Hi, please also update the docs https://github.com/tensorlayer/tensorlayer/blob/master/docs/modules/models.rst

and the changelog https://github.com/tensorlayer/tensorlayer/blob/master/CHANGELOG.md

**Comment by zsdonghao** [comment]
A better coding style can be found here https://github.com/tensorlayer/tensorlayer/blob/master/tensorlayer/models/squeezenetv1.py, instead of using `self.all_layers = net.xxxx`, we set the model as `Layer` class.
Therefore, for other models, you can connect tfslim into the layer class.

File: tensorlayer/models/vgg19.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+        net = net_in
+        for l in layers:
+            net = l(net)
+            # if end_with in net.name:
+            if net.name.endswith(end_with):
+                return net
+
+        raise Exception("unknown layer name (end_with): {}".format(end_with))
+
+    def restore_params(self, sess):
```
Comment (zsdonghao) [inline tensorlayer/models/vgg19.py]: we should make it download automatically. @lgarithm @luomai  any idea?

Comment (OwenLiuzZ) [reply tensorlayer/models/vgg19.py]: One simple solution is that we train all the models and store theses models in the libraryof the TL

Comment (luomai) [reply tensorlayer/models/vgg19.py]: I got a "This site can’t be reached" error when I tried to download the npy of VGG'19 from the website given. 

How large is the npy file? Can we host it in our pre-trained model repo?

Comment (OwenLiuzZ) [reply tensorlayer/models/vgg19.py]: Yes, it is about 570MB for this npy file. The public repo maximum  limit for the file is 100 MB. However, I do not know what the limit is in the private repo.

**Comment by lgarithm** [comment]
You can use ‘git lfs’ for uploading large file.

________________________________

**Comment by lgarithm** [comment]
@zsdonghao for vgg16 we used the pre-trained weights provided by <http://www.cs.toronto.edu/~frossard/post/vgg16/>. 
But there is no corresponding weights  for vgg19.
Where can we find a publicly available pre-trained vgg19 model for TF/TL?

**Comment by DEKHTIARJonathan** [comment]
@lgarithm https://github.com/tensorflow/models/tree/master/research/slim

**Comment by OwenLiuzZ** [comment]
@lgarithm I use the pre-trained model from this [repo](https://github.com/machrisaa/tensorflow-vgg)

File: tensorlayer/models/vgg19.py (hunk 1 above)
Comment (lgarithm) [reply tensorlayer/models/vgg19.py]: The original download link is for browser download only.
I've uploaded it to tensorlayer/pretrained-models, you can now use this download link in the program:

```
https://media.githubusercontent.com/media/tensorlayer/pretrained-models/master/models/vgg19.npy
```

**Comment by lgarithm** [comment]
Hi, @OwenLiuzZ I've tested this PR on a clean VM, it works!
Please resolve conflicts and then it will be ready to merge.

**Comment by OwenLiuzZ** [comment]
Hi, @lgarithm I have resolve the conflicts. Thanks for reminding.

**Pull request merged**